### PR TITLE
Stabilize monster emotions (anger/morale)

### PR DESCRIPTION
**Pull request by actual-nh**
#### Summary
Bugfixes "Stabilize monster emotions"

#### Purpose of change

Currently, one difficulty in creating monsters is coming up with ones that won't flee from the slightest hint of things they fear, but will be capable of fleeing if need be (won't be completely berserk later in a combat). (Some fixing admittedly is needed on the code regarding fleeing, but that is out of the scope of this PR.)

Another difficulty is monsters remaining angry far too long (thanks to something else intruding such as a zombie, usually); this is perhaps most notorious in the Forge of Wonder. While other fixes are also needed (such as more-targeted anger and, for the Aftershock Whately (sp?) cyberhound problem, some of the fixes in #50863), one part of the problem is that monster emotions do not return to their base points quickly enough, particularly after being pushed very far from these. Moreover, the code to return emotions to the base points does not take into account time outside the reality bubble.

#### Describe the solution

This PR:
- Sets limits on more anger triggers for how high they can push anger. In all cases, these are above 10, so attack will still be possible (I write "possible" due to the effects of morale).
- Prevents regen_morale from pushing anger and morale to *above* their base points.
- Speeds up emotions returning to their base points. Now, they will return at a rate of ~0.25 points per turn to no more than 15 away from their base points (15 being the largest trigger amount), and at a rate of ~0.125 points per turn when 15 or less away from their base points. (These are approximate due to the use of one_in( 4 ) and one_in( 2 ).)
- Enables catchup for emotion processing when brought back into the reality bubble for the basic emotional stabilization mentioned in the third of the above; regen_morale while outside the reality bubble may be managed later in this PR, or may have to wait for another PR.

As a side fix, it also makes summoned or similar monsters not attempt to strangle themselves (or else be unaffected by) leashes; I would appreciate if @eltank would take a look at this.

#### Describe alternatives you've considered

I have avoided the use of roll_remainder in the emotion processing catchup, by choosing to round the per-8-turn rates up if they are going in the direction of the monster calming down (lower fear, higher morale), with rounding down otherwise. The logic of this is that, while outside the reality bubble, they are not encountering the player, who is likely to be stirring quite a bit of the emotions in question.

Eventually, it would be nice to have one_in( # ) actually only randomize which turn something happens out of the # possible, but that is a matter for another PR.

#### Testing

This is going to require some extended playtesting (which I most definitely don't have time for; I can sort-of justify to myself programming and helping in other ways, but I have too many other things to do to not feel too guilty about playing!). Anyone familiar with current problems with monster emotions (e.g., @Venera3) is invited to take a look.

### CODE REVIEW

File: src/monster.cpp [hunk 1]
```diff
@@ -3146,6 +3150,51 @@ void monster::on_load()
...
+            dt_left_m -= adjust_by_m * 4;
+            if( morale < type->morale ) {
+                morale += adjust_by_m;
+            } else {
+                morale -= adjust_by_m;
+            }
+        }
+
+        if( morale <
+            type->morale ) { // Avoiding roll_remainder - PC out of situation, monster can calm down
```
Comment (actual-nh) [inline src/monster.cpp:3168]: I hadn't noticed astyle had done this; will fix.

**Comment by kevingranade** [comment]
Not *necessary* for this PR, but you're looking for calendar::once_every( time_point ), i.e. once_every( 4_turns ).

**Comment by eltank** [comment]
> As a side fix, it also makes summoned or similar monsters not attempt to strangle themselves (or else be unaffected by) leashes

Could you clarify under which circumstances this happens?

**Comment by actual-nh** [comment]
> > As a side fix, it also makes summoned or similar monsters not attempt to strangle themselves (or else be unaffected by) leashes
> 
> Could you clarify under which circumstances this happens? I'm willing to run a test, but I don't know how to reproduce the undesired behavior.

Summon a (friendly only due to the summons) monster, with nothing around for it to attack. Put a leash on it. Move away from it. (I haven't tested this myself, and a fix perhaps should not be in this PR, but it was in the middle of code that I've been working on and from what I'm seeing the critter won't follow you, despite the leash. Admittedly, @KorGgenT may prefer this behavior; I've got a lot to work on today, but will inquire.)

**Comment by eltank** [comment]
I tested it. Summoned animals (I used a wolf) cannot be leashed, either before or after this PR. There's no option in the menu.
That's because summoned animals use `monexamine::mfriend_menu`, not `monexamine::pet_menu`

**Pull request merged**